kanidm_book/src/integrations/radius.md

```diff
@@ -82,112 +82,189 @@ To read these secrets, the RADIUS server requires an account with the
-
-    [client.localhost]
-    ipaddr = 127.0.0.1
-    secret = testing123
-
-    [client.docker]
-    ipaddr = 172.17.0.0/16
-    secret = testing123
-
-You can then run the container with:
-
-    docker run --name radiusd -v ...:/data kanidm/radius:latest
+This container requires some cryptographic material, with the following files being in `/etc/raddb/certs`. (Modifiable in the configuration)
+   
+| filename   | description                                                   |
+| ---        | ---                                                           |
+| ca.pem     | The signing CA of the RADIUS certificate                      |
+| dh.pem     | The output of `openssl dhparam -in ca.pem -out ./dh.pem 2048` |
+| cert.pem   | The certificate for the RADIUS server                         |
+| key.pem    | The signing key for the RADIUS certificate                    |
+
+The configuration file (`/data/kanidm`) has the following template:
+
+```toml
+uri = "https://example.com" # URL to the Kanidm server
+verify_hostnames = true     # verify the hostname of the Kanidm server
+
+verify_ca = false           # Strict CA verification
+ca = /data/ca.pem           # Path to the kanidm ca
+username =                  # Username of the RADIUS service account
+password =                  # Generated secret for the service account
+
+# Default vlans for groups that don't specify one.
+radius_default_vlan = 1 
+
+# A list of Kanidm groups which must be a member
+# before they can authenticate via RADIUS.
+radius_required_groups = [
+	"radius_access_allowed",
+]
+
+# A mapping between Kanidm groups and VLANS
+radius_groups = [
+    { name = "radius_access_allowed", vlan = 10 },
+]
+
+# A mapping of clients and their authentication tokens
+radius_clients = [
+    { name = "test", ipaddr = "127.0.0.1", secret  = "testing123" },
+    # TODO: see if this works - it gets written out to the file
+    { name = "docker" , ipaddr = "172.17.0.0/16", secret = "testing123" },
+]
+
+# radius_cert_path = "/etc/raddb/certs/cert.pem"
+# the signing key for radius TLS
+# radius_key_path = "/etc/raddb/certs/key.pem"   
+# the diffie-hellman output
+# radius_dh_path = "/etc/raddb/certs/dh.pem"     
+# the CA certificate
+# radius_ca_path = "/etc/raddb/certs/ca.pem"     
+
+```
+
+## A fully configured example
+
+
+```toml
+url = "https://example.com"
+
+username = "radius_service_account"
+# The generated password from above
+password = "cr4bzr0ol" 
+
+# default vlan for groups that don't specify one.
+radius_default_vlan = 99 
+
+# if the user is in one of these Kanidm groups, 
+# then they're allowed to authenticate
+radius_required_groups = [
+    "radius_access_allowed",
+]
+
+radius_groups = [
+    { name = "radius_access_allowed", vlan = 10 }
+]
+
+radius_clients = [
+    { name = "localhost", ipaddr = "127.0.0.1", secret = "testing123" },
+    { name = "docker" , ipaddr = "172.17.0.0/16", secret = "testing123" },
+]
+```
+
+## Running a test RADIUS container
```

Comment:
Shouldn't the test radius be in the dev docs?

Reply:
Sure, can do that :smile